site/src/modules/resources/AgentLogs/useAgentLogs.test.tsx

```diff
@@ -0,0 +1,137 @@
+import { act, renderHook, waitFor } from "@testing-library/react";
+import WS from "jest-websocket-mock";
+import { type QueryClient, QueryClientProvider } from "react-query";
+import { API } from "api/api";
+import * as APIModule from "api/api";
+import { agentLogsKey } from "api/queries/workspaces";
+import type {
+  WorkspaceAgentLifecycle,
+  WorkspaceAgentLog,
+} from "api/typesGenerated";
+import { MockWorkspace, MockWorkspaceAgent } from "testHelpers/entities";
+import { createTestQueryClient } from "testHelpers/renderHelpers";
+import { type UseAgentLogsOptions, useAgentLogs } from "./useAgentLogs";
+
+afterEach(() => {
+  WS.clean();
+});
+
+describe("useAgentLogs", () => {
+  it("should not fetch logs if disabled", async () => {
+    const queryClient = createTestQueryClient();
+    const fetchSpy = jest.spyOn(API, "getWorkspaceAgentLogs");
+    const wsSpy = jest.spyOn(APIModule, "watchWorkspaceAgentLogs");
+    renderUseAgentLogs(queryClient, "ready", { enabled: false });
+    expect(fetchSpy).not.toHaveBeenCalled();
+    expect(wsSpy).not.toHaveBeenCalled();
+  });
+
+  it("should return existing logs without network calls", async () => {
+    const queryClient = createTestQueryClient();
+    queryClient.setQueryData(
+      agentLogsKey(MockWorkspace.id, MockWorkspaceAgent.id),
+      generateLogs(5),
+    );
+    const fetchSpy = jest.spyOn(API, "getWorkspaceAgentLogs");
+    const wsSpy = jest.spyOn(APIModule, "watchWorkspaceAgentLogs");
+    const { result } = renderUseAgentLogs(queryClient, "ready");
+    await waitFor(() => {
+      expect(result.current).toHaveLength(5);
+    });
+    expect(fetchSpy).not.toHaveBeenCalled();
+    expect(wsSpy).not.toHaveBeenCalled();
+  });
+
+  it("should fetch logs when empty", async () => {
```

Comment:
Just for clarity, could the description be updated to mention that there are no requests expected if the lifecycle isn't `starting`, even if we do have data?